airbyte-cdk/python/airbyte_cdk/sources/declarative/manifest_declarative_source.py

```diff
@@ -142,6 +144,33 @@ def _validate_source(self):
         except ValidationError as e:
             raise ValidationError("Validation against json schema defined in declarative_component_schema.yaml schema failed") from e
 
+        cdk_version = metadata.version("airbyte_cdk")
+        cdk_major, cdk_minor, cdk_patch = (int(part) for part in cdk_version.split("."))
+        manifest_version = self._source_config.get("version")
+        manifest_major, manifest_minor, manifest_patch = self._get_version_parts(manifest_version)
+
+        if cdk_major < manifest_major or (cdk_major == manifest_major and cdk_minor < manifest_minor):
```

Comment:
why not checking the patch version here?

Reply:
The idea here is that if we're sticking with semantic versioning, the patch should always be backwards compatible and that we should be encouraging or supporting use of the latest patch since it has fixes. I'm happy to add it, but I also feel like we shouldn't be strict on patches

Reply:
Fair point, if we make sure to never add functionality in a patch version it should be fine